File: snippets/python/create_a_new_campaign.py
import os
import mailchimp_marketing as MailchimpMarketing
from mailchimp_marketing.api_client import ApiClientError

try:
  client = MailchimpMarketing.Client()
  client.setConfig(os.getenv("API_KEY"), os.getenv("API_SERVER"))
  response = client.campaigns.create({"type": "plaintext"})
  print("client.ping.get() response: {}".format(response))
except ApiClientError as error:
  print("An exception occurred: {}".format(error.text))

File: snippets/ruby/create_a_new_campaign.rb
require 'MailchimpMarketing'

begin
  client = MailchimpMarketing::Client.new
  client.set_config(ENV['API_KEY'], ENV['API_SERVER'])
  result = client.campaigns.create({ 'type' => 'variate' })
  p result
rescue MailchimpMarketing::ApiError => e
  puts "Error: #{e}"
end
